base/linalg/lu.jl

```diff
@@ -179,36 +179,13 @@ function logabsdet{T,S}(A::LU{T,S})  # return log(abs(det)) and sign(det)
         end
         abs_det += log(abs(dg_ii))
     end
-    s = (isodd(c) ? -one(real(T)) : one(real(T))) * P
+    s = (isodd(c) ? -one(T) : one(T)) * P
```

Comment:
After using 'real' like you suggested, the test cases didn't pass.
 ```julia
   * linalg/lu            Error During Test
  Test threw an exception of type ErrorException
  Expression: @inferred(logdet($(Expr(:typed_vcat, :Complex64, :(1.0f0 0.5f0), :(0.5f0 -1.0f0))))) === 0.22314355f0 + 3.1415927f0im
  return type Complex{Float32} does not match inferred return type Union{Complex{Float32},Float32}
```
To be more precise this failed:
```julia
@test @inferred(logdet(Complex64[1.0f0 0.5f0; 0.5f0 -1.0f0])) === 0.22314355f0 + 3.1415927f0im
```